src/Couchbase.Lite.Shared/API/Sync/ReplicatorConfiguration.cs

```diff
@@ -140,7 +142,9 @@ public bool Continuous
         /// Gets the local database participating in the replication. 
         /// </summary>
         [NotNull]
-        public Database Database { get; }
+        public Database Database => Collections.Count > 0 ? Collections[0].Database 
+            ?? throw new CouchbaseLiteException(C4ErrorCode.InvalidParameter, "Database is missing in Replication Configuration.")
+            : throw new CouchbaseLiteException(C4ErrorCode.InvalidParameter, "Database is missing in Replication Configuration.");
```

Comment:
This is messy to look at, I think you can make it cleaner with something like `Collections.Select(x => x.Database).FirstOrDefault(d => d != null) ?? throw`

Reply:
Or simply `Collections.Count > 0 && Collections[0].Database != null ? Collections[0].Database : throw`